test/local/account_routes.js

```diff
@@ -213,3 +213,146 @@ test(
     })
   }
 )
+
+test(
+  'device updates dont write to the db if nothing has changed',
+  function (t) {
+    var uid = uuid.v4('binary')
+    var deviceId = crypto.randomBytes(16)
+    var mockRequest = {
+      auth: {
+        credentials: {
+          uid: uid.toString('hex'),
+          deviceId: deviceId,
+          deviceName: 'my awesome device',
+          deviceType: 'desktop',
+          deviceCallbackURL: '',
+          deviceCallbackPublicKey: '',
+        }
+      },
+      payload: {
+        id: deviceId.toString('hex'),
+        name: 'my awesome device'
+      }
+    }
+    var mockDB = {
+      updateDevice: sinon.spy(function () {
+        return P.resolve()
+      })
+    }
+    var mockLog = mocks.spyLog()
+    var accountRoutes = makeRoutes({
+      db: mockDB,
+      log: mockLog
+    })
+    return new P(function(resolve) {
+      getRoute(accountRoutes, '/account/device')
+        .handler(mockRequest, function(response) {
+          resolve(response)
+        })
+    })
+    .then(function(response) {
+      t.equal(mockDB.updateDevice.callCount, 0, 'updateDevice was not called')
+
+      t.equal(mockLog.increment.callCount, 1, 'a counter was incremented')
+      t.equal(mockLog.increment.firstCall.args[0], 'device.update.spurious')
```

Comment:
Wow, I never noticed these local route tests before, are they new? I should have used them for the KPI activty event stuff, will make sure I do so from now on.